internal/controller/serverset/serverset_crossplane_mock_test.go

```diff
@@ -0,0 +1,242 @@
+	"github.com/crossplane/crossplane-runtime/pkg/meta"
+	"github.com/crossplane/crossplane-runtime/pkg/reconciler/managed"
+	"github.com/crossplane/crossplane-runtime/pkg/resource"
+	"github.com/crossplane/crossplane-runtime/pkg/resource/fake"
+	"github.com/crossplane/crossplane-runtime/pkg/test"
+	"github.com/stretchr/testify/assert"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	"github.com/ionos-cloud/crossplane-provider-ionoscloud/apis/compute/v1alpha1"
+	"github.com/ionos-cloud/crossplane-provider-ionoscloud/pkg/kube"
+)
+
+func Test_serverSet_Create(t *testing.T) {
+	fakeObjList := func(obj client.ObjectList) error { return nil }
+	fakeGetObj := func(obj client.Object) error {
+		switch res := obj.(type) {
+		case *v1alpha1.Server:
+			res.Status.AtProvider.State = "AVAILABLE"
+			res.Status.AtProvider.ServerID = "test-id"
+		case *v1alpha1.Volume:
+			res.Status.AtProvider.State = "AVAILABLE"
+			res.Status.AtProvider.VolumeID = "test-id"
+		case *v1alpha1.Nic:
+			res.Status.AtProvider.State = "AVAILABLE"
+			res.Status.AtProvider.NicID = "test-id"
+		}
+		return nil
+	}
+	type fields struct {
+		kube                 client.Client
+		mg                   resource.Managed
+		bootVolumeController kubeBootVolumeControlManager
+		nicController        kubeNicControlManager
+		serverController     kubeServerControlManager
+		log                  logging.Logger
+	}
+	type args struct {
+		ctx context.Context
+		cr  *v1alpha1.ServerSet
+	}
+
+	tests := []struct {
+		name    string
+		fields  fields
+		args    args
+		want    managed.ExternalCreation
+		wantErr error
+	}{
+		{
+			name: "server set successfully created",
+			fields: fields{
+				kube: &test.MockClient{
+					MockList: test.NewMockListFn(nil, fakeObjList),
+				},
+				log: logging.NewNopLogger(),
+				mg:  &fake.Managed{},
+				bootVolumeController: &kubeBootVolumeController{
+					kube: &test.MockClient{
+						MockCreate: test.NewMockCreateFn(nil),
+						MockList:   test.NewMockListFn(nil, fakeObjList),
+						MockGet:    test.NewMockGetFn(nil, fakeGetObj),
+					},
+					log: logging.NewNopLogger(),
+				},
+				serverController: &kubeServerController{
+					kube: &test.MockClient{
+						MockCreate: test.NewMockCreateFn(nil),
+						MockList:   test.NewMockListFn(nil, fakeObjList),
+						MockGet:    test.NewMockGetFn(nil, fakeGetObj),
+					},
+					log: logging.NewNopLogger(),
+				},
+				nicController: &kubeNicController{
+					kube: &test.MockClient{
+						MockCreate: test.NewMockCreateFn(nil),
+						MockList:   test.NewMockListFn(nil, fakeObjList),
+						MockGet:    test.NewMockGetFn(nil, fakeGetObj),
+					},
+					log: logging.NewNopLogger(),
+				},
+			},
+			args: args{
+				ctx: context.Background(),
+				cr:  createBasicServerSet(),
+			},
+			want: managed.ExternalCreation{
+				ConnectionDetails: managed.ConnectionDetails{},
+			},
+			wantErr: nil,
+		},
+		{
+			name: "failed to create, too many volumes",
```

Comment:
We already have a test for this scenario: https://github.com/ionos-cloud/crossplane-provider-ionoscloud/blob/feat/serverset_ctrl/internal/controller/serverset/serverset_test.go#L662